### PR TITLE
Preview card contents

### PR DESCRIPTION
**Pull request by hosseintoussi**
In order to have a nice preview, I am removing coderay which handles the
syntax highlighting on the server side to highlighjs which takes care
of highlighting on the client side. 

With this I am ensuring that we only have one way of highlighting syntaxes.

In terms of markdown, server side remains using redcarpet, but frontend
for previews uses MarkedJS.

### CODE REVIEW

File: app/assets/javascripts/preview.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    gfm: true,
+    tables: true,
+    breaks: true,
+    pedantic: false,
+    sanitize: false,
+    smartLists: true,
+    smartypants: false
+  });
+
+  var text = $("#card-description").val();
```
Comment (nomorejay) [inline app/assets/javascripts/preview.js]: if you wanted to be cute and remove the if statement hinge for pure logic you could write it like this:
$(".preview").html(text && marked(text) ||  "Nothing to preview...")

but maybe that's, what they call in bird culture, a dick move

Comment (ricardovaleriano) [inline app/assets/javascripts/preview.js]: Yup.
Or even live the `if` but extract the `html` call could be beneficial for the reading? Or is just a matter of taste?

```js
preview = text ? makred(text) : "Nothing to preview";
$(".preview").html(preview);
```

Comment (nomorejay) [inline app/assets/javascripts/preview.js]: ya ricardo's idea is the best solution for readability and simplicity.

Comment (hosseintoussi) [inline app/assets/javascripts/preview.js]: awesome.. yea makes a lot of sense.

**Comment by hosseintoussi** [comment]
@nomorejay @ricardovaleriano thanks a lot guys. :)

**Pull request merged**